source/mir/stat/descriptive.d

```diff
@@ -4696,3 +4696,1463 @@ unittest
     assert(x.sliced.coefficientOfVariation.approxEqual(2.231299 / 2.437500));
     assert(x.sliced.coefficientOfVariation!float.approxEqual(2.231299 / 2.437500));
 }
+
+///
+struct MomentAccumulator(T, size_t N, Summation summation)
+    if (N > 0 && isMutable!T)
+{
+    import std.traits: isIterable;
+
+    ///
+    Summator!(T, summation) summator;
+
+    ///
+    size_t count;
+
+    ///
+    F moment(F = T)() const @safe @property pure nothrow @nogc
+    {
+        return cast(F) summator.sum / cast(F) count;
+    }
+
+    ///
+    F sumOfPower(F = T)() const @safe @property pure nothrow @nogc
+    {
+        return cast(F) summator.sum;
+    }
+
+    ///
+    void put(Range)(Range r)
+        if (isIterable!Range)
+    {
+        import mir.math.func.powi: powi;
+        import mir.primitives: hasShape;
+
+        static if (hasShape!Range)
+        {
+            import mir.ndslice.topology: map;
+            import mir.primitives: elementCount;
+
+            count += r.elementCount;
+            summator.put(r.map!(a => a.powi(N)));
+        }
+        else
+        {
+            foreach(x; r)
+            {
+                put(x);
+            }
+        }
+    }
+
+    ///
+    void put(Range)(Range r, T m)
+        if (isIterable!Range)
+    {
+        import mir.math.func.powi: powi;
+        import mir.primitives: hasShape;
+
+        static if (hasShape!Range)
+        {
+            import core.lifetime: move;
+            import mir.ndslice.internal: LeftOp;
+            import mir.ndslice.topology: vmap, map;
+            import mir.primitives: elementCount;
+
+            count += r.elementCount;
+            summator.put(r.move.
+                    vmap(LeftOp!("-", T)(m)).
+                    map!(a => a.powi(N))
```

Comment:
There should be two specializations for N == 1 and N == 2

Reply:
Added